File: internal/ports/repository/environment.go
package repository

import (
	"context"

	"github.com/elouan/dockyard/internal/domain"
)

type EnvironmentSetRepository interface {
	List(ctx context.Context, projectID string) ([]domain.EnvironmentSet, error)
	Create(ctx context.Context, set domain.EnvironmentSet) (domain.EnvironmentSet, error)
	GetByID(ctx context.Context, id string) (domain.EnvironmentSet, error)
	GetDefaultForProject(ctx context.Context, projectID string) (domain.EnvironmentSet, error)
}

type EnvironmentVariableRepository interface {
	ListBySet(ctx context.Context, environmentSetID string) ([]domain.EnvironmentVariable, error)
	Upsert(ctx context.Context, environmentSetID, key, value string, isSecret bool) error
	Delete(ctx context.Context, id string) error
}

File: web/lib/types/environment.ts
export interface EnvironmentSet {
  id: string;
  projectId: string;
  name: string;
}

export interface EnvironmentVariable {
  id: string;
  environmentSetId: string;
  key: string;
  value: string;
  isSecret: boolean;
}

export interface CreateEnvironmentSetPayload {
  name: string;
}

export interface UpsertVariablePayload {
  key: string;
  value: string;
  isSecret: boolean;
}
